### PR TITLE
Update story lifecycle to reflect current+proposed practices

### PR DESCRIPTION
**Pull request by mogul**
I'm leaving this PR in a Draft state until people have had a chance to review and react to it, and will merge it if there's no more chatter on it after 3 weekdays (EOB Monday May 5 2022).

## Changes proposed in this pull request:
- Updates existing story lifecycle documentation to reflect evolved practices and ideals

## security considerations
None, this is a team process document rather than code, and this information is deliberately open for transparency and to share learning with other teams.

### CODE REVIEW

**Comment by rcgottlieb** [comment]
By "explicitly identified" columns, you mean that there could be more columns, but these are the minimum any given board should have, correct?

**Comment by mogul** [comment]
> By "explicitly identified" columns, you mean that there could be more columns, but these are the minimum any given board should have, correct?

Yes, that's some careful language there about the board "including" those columns, but not saying that's _all_ of them. That gives us a lot more flexibility with how we organize and triage stuff on the board that's not actually getting effort yet.

**Comment by mogul** [comment]
From [Slack](https://gsa-tts.slack.com/archives/GNSL334HM/p1651158002517319?thread_ts=1651120751.295629&cid=GNSL334HM):
> [Robert Gottlieb]
> [...] I think it’s probably fine the way it is. I just don’t know how detailed it needs to be.

We have to strike a balance between "useful to us and specific enough to act as a compliance artifact" and "not so specific that we have no flexibility to deviate without assessors calling it a finding".

File: StoryLifecycle.md [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
+  * We default to specifying ACs using [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development#Behavioral_specifications) to make it easier to demonstrate the outcome of the work.
+* Include a reviewed "Security Considerations" section. Especially ensure any authentication/authorization and data persistence points are discussed and requirements are addressed via Acceptance Criteria (e.g., "Authenticate using [login.fr.cloud.gov]" or "Any ephemeral data resulting from usage is backed up/recoverable").
+* Be reasonably small. 
+  * If we agree a story is likely to take more than two weeks of concentrated effort to complete, we [split the story into smaller, individually-valuable stories](https://www.humanizingwork.com/the-humanizing-work-guide-to-splitting-user-stories/) and prioritize them independently.
 
 Required for compliance:
 
 * If necessary, the story includes a security testing plan. For example, the ACs include automated tests and alerts for unexpected behavior.
+<!-- @mogul says: It's not clear to me why all this feature-level language is in the StoryLifecycle document; we already have it in the FeatureLifecycle document! -->
```
Comment (mogul) [inline StoryLifecycle.md:23]: @pburkholder can you comment on this? I don't know if it's in here intentionally or as a copy-and-paste leftover.

**Comment by mogul** [comment]
> This is nitpicky, but can we replace our use of the term `groomed` with `refined` or something similar?

Done.

**Comment by mogul** [comment]
I just realized that if you're scanning the diff it may look like I changed a lot more content than I did... Most of the big diffs are just indentation/formatting. Maybe use [this link](https://github.com/cloud-gov/product/pull/1739/files?short_path=556efb6#diff-556efb60ae6137ea283ac1675b37131c5632e4f31fa255a2d68d458e490aaa9a) to review the rich-diff so you can concentrate on the actual content changes. (Note to self: Next time put content and formatting updates in different commits.)

**Comment by mogul** [comment]
Final call! Planning to merge this later today if there are no further comments.

**Comment by pburkholder** [comment]
I need to look at this and I have not slotted any time for this today.

**Pull request merged**